src/custom_modules/DAO.ts

```diff
@@ -69,24 +87,50 @@ export const DAO = {
     cb();
   },
 
-  query: async function <T>(query, params = undefined, retries = 0) {
+  query: async function <T = any>(
+    query,
+    params = undefined,
+    retries = 0,
+  ): Promise<
+    [
+      T extends Array<Record<string, any>>
+        ? RawPrismaTypes<T>
+        : T extends Record<string, any>
+        ? RawPrismaType<T>
+        : T,
+      any,
+    ]
+  > {
     try {
-      return await this.dbPool.query(query, params);
+      return await (this as typeof DAO).dbPool.query<any>(query, params);
```

Comment:
Seems like a smell that we have to do this assertion. Would simplify things to skip the DAO object right? Not saying we should do that now though :slightly_smiling_face:

Reply:
Somehow the type of this is not inferred, hence the assertion. I'm not sure why though 😋